.prettierignore

```diff
@@ -52,3 +52,10 @@ docs/code-docs/
 
 # prettier-plugin-sql does not properly support sprocs, so ignoring those
 apps/prairielearn/src/sprocs/*.sql
+
+# exampleCourse and testCourse contain info.json files with different formatting on purpose, ignore
+exampleCourse/**/*.json
+testCourse/**/*.json
+
+# JSON files in tests are formatted explicitly to check for loading behaviour, so should not be formatted automatically
```

Comment:
Interesting, I don't actually know what this is referring to. Can you link to a test that depends on formatting?

Reply:
One example is: https://github.com/PrairieLearn/PrairieLearn/blob/993e20be4503c0be5576b307036604541f052c39/apps/prairielearn/src/tests/fileEditor.test.js#L539

The file editor test changes a file in the test course, and checks if the change is properly applied by comparing the contents of the JSON file with the stringified representation of the content. I could change this line to use 2 instead of 4 spaces, but this still fails the test:

```
  864 passing (10s)
  1 failing

  1) test file editor
       not the test course
         verify edits
           edit questions/test/question/info.json
             GET to edit url
               should match expected draft file contents:

      AssertionError: expected '{\n  "uuid": "209aa5b5-cce7-4c9a-9252…' to equal '{\n  "uuid": "209aa5b5-cce7-4c9a-9252…'
      + expected - actual

       {
         "uuid": "209aa5b5-cce7-4c9a-9252-00823d8494a9",
         "title": "Test question",
         "topic": "Test",
      -  "tags": ["tbretl"],
      +  "tags": [
      +    "tbretl"
      +  ],
         "type": "v3"
       }

      at Context.<anonymous> (src/tests/fileEditor.test.js:638:12)
      at processImmediate (node:internal/timers:466:21)
      at process.topLevelDomainCallback (node:domain:161:15)
      at process.callbackTrampoline (node:internal/async_hooks:128:24)
```

While this could potentially require a change in the test to ensure it checks the parsed content instead of the formatted content, I believe this is better done (if needed) in a future separate PR.

Reply:
Another case is
https://github.com/PrairieLearn/PrairieLearn/blob/993e20be4503c0be5576b307036604541f052c39/apps/prairielearn/src/tests/testJsonLoad/broken.json#L2

where a trailing comma is added on purpose to trigger an error on loading. Since prettier removes the comma, the error is no longer there and a test that expects an error no longer sees it.

Reply:
Makes sense, thanks for linking to those! Agreed that it's fine to skip formatting these files entirely.